onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -642,12 +692,32 @@ Status ReduceKernel<true>::ComputeImpl<int32_t, CUDNN_REDUCE_TENSOR_NO_INDICES>(
   typedef typename ToCudaType<int32_t>::MappedType CudaT;
 
   const Tensor* X = ctx->Input<Tensor>(0);
+  std::vector<int64_t> axes;
+
+  size_t num_inputs = ctx->InputCount();
+  if (num_inputs == 2) {
+    //override the attribute value with the input value for reduction_axes
+    const Tensor* axes_tensor = ctx->Input<Tensor>(1);
+    ORT_ENFORCE(axes_tensor != nullptr, "Axes input is null");
+    ORT_ENFORCE(axes_tensor->Shape().NumDimensions() == 1, "An axes tensor must be a vector tensor.");
+    auto nDims = static_cast<size_t>(axes_tensor->Shape()[0]);
+    const auto* data = axes_tensor->template Data<int64_t>();
+    axes.assign(data, data + nDims);
+    // empty axes and no-op
+    if (axes.empty() && noop_with_empty_axes_) {
+      auto* Y = ctx->Output(0, X->Shape());
+      CUDA_RETURN_IF_ERROR(cudaMemcpyAsync(Y->template MutableData<int32_t>(), X->template Data<int32_t>(), X->SizeInBytes(), cudaMemcpyDeviceToDevice));
+      return Status::OK();
+    }
```

Comment:
Can we move the empty axes handling outside the if check

Reply:
it is only supported for opset 13 and above, but changing since the functionality should be independent of how axes is read.